apps/api/src/bid/controllers/bid/bid.controller.ts

```diff
@@ -0,0 +1,36 @@
+import { BidHttpService } from "@akashnetwork/http-sdk";
+import assert from "http-assert";
+import flatten from "lodash/flatten";
+import { singleton } from "tsyringe";
+
+import { AuthService, Protected } from "@src/auth/services/auth.service";
+import { ListBidsResponse } from "@src/bid/http-schemas/bid.schema";
+import { UserWalletRepository } from "@src/billing/repositories";
+
+@singleton()
+export class BidController {
+  constructor(
+    private readonly bidHttpService: BidHttpService,
+    private readonly authService: AuthService,
+    private readonly userWalletRepository: UserWalletRepository,
+  ) {}
+
+  @Protected([{ action: "create", subject: "DeploymentSetting" }])
+  async list(dseq: string, userId?: string): Promise<ListBidsResponse> {
+    const { currentUser } = this.authService;
+    assert(this.isValidUserId(userId), 401);
+
+    const wallets = await this.userWalletRepository.findByUserId(userId ?? currentUser.userId);
```

Comment:
we better use permissions here and remove the check above.

```suggestion
    const wallets = await this.userWalletRepository.accessibleBy(ability, "sign").findByUserId(userId ?? currentUser.userId);
```

Reply:
We can add the accessibleBy part, added it now.
But: if we remove the decortor call, we'll have to manually check `ability.can("sign", "UserWallet")`, for an early 401. I would rather keep the decorator version instead of an assert call inside, because I think a decorator on top of the method is an easier read next time. What do you think?

Reply:
Yea, I think decorator is fine, I'm referring to the assert

Reply:
Ah, that one! You're right, it's not really needed after we have `.accessibleBy`, removed it.